src/RazorSdk/Targets/Microsoft.NET.Sdk.Razor.SourceGenerators.targets

```diff
@@ -29,13 +29,28 @@ Copyright (c) .NET Foundation. All rights reserved.
       <EditorConfigFiles Include="$(RazorEncConfigFile)" Condition="'$(DesignTimeBuild)' == 'true' AND '$(BuildingInsideVisualStudio)' == 'true'"/>
     </ItemGroup>
 
-    <!-- Configure analyzers -->
-    <ItemGroup>
+    <!-- Assemblies in the SDK are cross-genned to improve performance. However, these
+    cross-genned assemblies cannot be loaded into 32-bit VS. To work around these, we load
+    the binaries that are not cross-genned (those in the source-generators directory) when
+    building in VS. However, we want to continue to leverage the cross-genned assemblies in
+    environments where it makes sense so we load them in the CLI and elsewhere.
+    For more info, see https://github.com/dotnet/aspnetcore/issues/32296. -->
+    <ItemGroup Condition="'$(BuildingInsideVisualStudio)' == 'true'">
```

Comment:
```suggestion
    <ItemGroup Condition="'$(MSBuildRuntimeType)' != 'Core'">
```

Reply:
Perhaps that's a better proxy for what we're trying to find i.e. is the compiler this will eventually get run in have the same  bitness as the SDK?

Reply:
Ah, I wasn't familiar with this property. Yeah, this is definitely a better way of conveying what we want.